packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts

```diff
@@ -126,11 +125,10 @@ function addV2Swap(
   // V2 trade should have only one route
   const route = trade.routes[0]
   const path = route.path.map((token) => token.wrapped.address)
-  const recipient = routerMustCustody
-    ? ROUTER_AS_RECIPIENT
-    : typeof options.recipient === 'undefined'
-    ? SENDER_AS_RECIPIENT
-    : validateAndParseAddress(options.recipient)
+
+  if (!options.recipient) throw new Error('recipient is required')
```

Comment:
I think we should accept undefined for recipient? By default should send the fund back to the caller?

Reply:
we better either it as required by extending the existing type `PancakeSwapOptions`, or we allow it to be undefined and use `SENDER_AS_RECIPIENT` by default

Reply:
default value is assigned by outside

https://github.com/pancakeswap/pancake-frontend/blob/ebf08dcd8234646f7c6488297178cc3afc049d2e/packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts#L52